packages/hooks/src/hooks/useEffectAfterInitialRender.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useRef } from "react";
+
+/**
+ * Like React's useEffect, except doesn't execute on initial render.
+ */
+export const useEffectAfterInitialRender = (
+  func: () => void,
```

Comment:
This doesn't properly wrap the `useEffect` type, as the return value of the effect is an optional callable for cleaning up whatever the effect did, which you'd want to return.

Reply:
I did some changes based on this comment. I'm not 100% they are now correct, i.e. I wasn't sure whether the returned cleanup should also do something for the ref-value used by the hook internally. Things seemed to work as expected without additional tampering, so I just return whatever the effect-function returns.

Reply:
Understood, let's go with this for now then. That said, NextJS imposing restrictions on non-NextJS components is a design consideration we might need to evaluate a bit more down the line, it seems like it could end up being counter productive even if the alternative is duplicating a bit of code.